### PR TITLE
feat: support file-logger plugin of response body record

### PR DESCRIPTION
**Pull request by pixeldin**
### Description

Providing some optional setting items for **file-logger** to control the output log content as requested in https://github.com/apache/apisix/issues/8331.

I refer some options of the [http-logger](https://apisix.apache.org/docs/apisix/next/plugins/http-logger/): 
`include_req_body`, `include_resp_body` and `include_resp_body_expr`
### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/file-logger2.t [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    if (! $block->request) {
+        $block->set_value("request", "GET /t");
+        if (!$block->response_body) {
+            $block->set_value("response_body", "passed\n");
+        }
+    }
+
+    if (! $block->no_error_log && ! $block->error_log) {
+        $block->set_value("no_error_log", "[error]");
+    }
```
Comment (tzssangglass) [inline t/plugin/file-logger2.t:34]: we don't need it, it's already set by default in APISIX.pm

Comment (pixeldin) [reply t/plugin/file-logger2.t:34]: Ok, thanks

**Comment by tzssangglass** [comment]
hi @pixeldin , we have error log as ` attempt to set ngx.status after sending out response headers`, we need to avoid call `ngx.status = code` after `ngx.say`, try with 

```diff
diff --git t/plugin/file-logger2.t t/plugin/file-logger2.t
index ccd1d9f5..ad4396c0 100644
--- t/plugin/file-logger2.t
+++ t/plugin/file-logger2.t
@@ -164,7 +164,6 @@ contain with target
 
             if new_msg.response.body == "hello world\n"
             then
-                ngx.status = code
                 ngx.say('contain target body hits with expr')
             end
 
@@ -174,7 +173,6 @@ contain with target
             local new_msg = core.json.decode(msg)
             if new_msg.response.body == nil
             then
-                ngx.status = code
                 ngx.say('skip unconcern body')
             end
         }

```

File: t/plugin/file-logger2.t [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+            if new_msg.response.body == nil
+            then
+                ngx.status = code
+                ngx.say('skip unconcern body')
+            end
+        }
+    }
+--- response_body
+skip unconcern body
```
[the remark this answers is not in the document]
Comment (tzssangglass) [reply t/plugin/file-logger2.t]: I'm not saying remove TEST 5 and TEST 6, just don't need TEST 5 and change `t("/hello?foo=bar", ngx.HTTP_GET)` to  `t("/hello?foo=jsck", ngx.HTTP_GET)` in TEST 6.

[the remark this answers is not in the document]
Comment (pixeldin) [reply t/plugin/file-logger2.t]: Hello @tzssangglass, thanks for your consider advice, I merge two different case into TEST 4, which show the case where the expression hits and the case that should be skipped. Is it all case covered, right?

File: t/plugin/file-logger2.t [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            end
+
+            --- a new request is logged
+            t("/hello?name=pix", ngx.HTTP_GET)
+            msg = fd:read("*l")
+            local new_msg = core.json.decode(msg)
+            if new_msg.response.body == nil
+            then
+                ngx.say('skip unconcern body')
+            end
```
Comment (tzssangglass) [inline t/plugin/file-logger2.t:181]: here the `msg` is read by `msg = fd:read()` before the new request is sent, so it's not possible for it to record new requests about it?

I suggest writing the new request as a new test case.

Comment (pixeldin) [reply t/plugin/file-logger2.t:181]: Thanks for your follow, here's the case:
Request-1, `fd:read()` read the first line content,
Request-2, `fd:read("*l")` read the next line.
What the log file generates:
```
1 {"upstream":"127.0.0.1:1982","service_id":"","server":{"hostname":"pixelpig","version":"3.0.0"},"request":{"querystring":{"foo":"bar"},"headers":{"content-type":"application\/x-www-form-urlencoded","user-agent":"lua-resty-http\/0.16.1 (Lua) ngx_lua\/10021","host":"127.0.0.1:1984"},"method":"GET","size":157,"uri":"\/hello?foo=bar","url":"http:\/\/127.0.0.1:1984\/hello?foo=bar"},"apisix_latency":100.99994659424,"start_time":1670294090157,"route_id":"1","upstream_latency":5,"latency":105.99994659424,"response":{"size":123,"headers":{"server":"APISIX\/3.0.0","content-type":"text\/plain","content-length":"12","connection":"close"},"body":"hello world\n","status":200},"client_ip":"127.0.0.1"}
2 {"upstream":"127.0.0.1:1982","service_id":"","server":{"hostname":"pixelpig","version":"3.0.0"},"request":{"querystring":{"name":"pix"},"headers":{"content-type":"application\/x-www-form-urlencoded","user-agent":"lua-resty-http\/0.16.1 (Lua) ngx_lua\/10021","host":"127.0.0.1:1984"},"method":"GET","size":158,"uri":"\/hello?name=pix","url":"http:\/\/127.0.0.1:1984\/hello?name=pix"},"apisix_latency":100.00016403198,"start_time":1670294090274,"route_id":"1","upstream_latency":3,"latency":103.00016403198,"response":{"headers":{"server":"APISIX\/3.0.0","content-type":"text\/plain","content-length":"12","connection":"close"},"size":123,"status":200},"client_ip":"127.0.0.1"}
```
With the read cursor moving, that's why the branch `if new_msg.response.body == nil` will hit, Is my understanding right?

Comment (tzssangglass) [reply t/plugin/file-logger2.t:181]: yes, my mistake.

**Review by tzssangglass — APPROVED** [review]

File: t/plugin/file-logger2.t [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+            -- note only for first line
+            msg = fd:read()
+
+            local new_msg = core.json.decode(msg)
+            ngx.status = code
+
+            if new_msg.response ~=nil and new_msg.response.body == "hello world\n"
+            then
```
[the remark this answers is not in the document]
Comment (pixeldin) [reply t/plugin/file-logger2.t:169]: ok, thanks

File: apisix/plugins/file-logger.lua [hunk 5]
```diff
@@ -31,6 +31,15 @@ local schema = {
         path = {
             type = "string"
         },
+        include_req_body = {type = "boolean", default = false},
```
[the remark this answers is not in the document]
Comment (pixeldin) [reply apisix/plugins/file-logger.lua:34]: No, I will remove it.

**Review by tzssangglass — DISMISSED** [review]

**Comment by pixeldin** [comment]
@tzssangglass please retrigger the CI workflow, the running state seems unstable.

**Comment by spacewander** [comment]
> @tzssangglass please retrigger the CI workflow, the running state seems unstable.

Done

File: t/plugin/file-logger2.t [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+            end
+
+            -- note only for first line
+            msg = fd:read()
+
+            local new_msg = core.json.decode(msg)
+            ngx.status = code
+
+            if new_msg.response ~= nil and new_msg.response.body == "hello world\n"
+            then
```
Comment (spacewander) [inline t/plugin/file-logger2.t:96]: The `then` should be the same line with `if`, just like what we have done in other code

File: t/plugin/file-logger2.t [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+                return
+            end
+
+            -- note only for first line
+            msg = fd:read()
+
+            local new_msg = core.json.decode(msg)
+            ngx.status = code
+
+            if new_msg.response ~= nil and new_msg.response.body == "hello world\n"
```
Comment (spacewander) [inline t/plugin/file-logger2.t:168]: Ditto

File: t/plugin/file-logger2.t [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+                ngx.status = code
+                ngx.say('contain target body hits with expr')
+            end
+
+            --- a new request is logged
+            t("/hello?name=pix", ngx.HTTP_GET)
+            msg = fd:read("*l")
+            local new_msg = core.json.decode(msg)
+            if new_msg.response.body == nil
+            then
```
Comment (spacewander) [inline t/plugin/file-logger2.t:179]: Ditto

File: t/plugin/file-logger2.t (hunk 7 above)
Comment (pixeldin) [reply t/plugin/file-logger2.t:168]: ok. thanks~

**Review by tzssangglass — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**